noir-projects/aztec-nr/aztec/src/note/utils.nr

```diff
@@ -67,6 +67,16 @@ pub fn compute_note_hash_for_insertion<Note, N>(note: Note) -> Field where Note:
     compute_inner_note_hash(note)
 }
 
+pub fn compute_note_hash_for_read_request<Note, N>(note: Note) -> Field where Note: NoteInterface<N> {
+    let header = note.get_header();
+
+    if (header.nonce != 0) {
+        compute_unique_note_hash(note)
+    } else {
+        compute_inner_note_hash(note)
+    }
+}
```

Comment:
@LeilaWang I am just profiling circuits and this if else makes this quite inefficient since if-elses are not real (see https://github.com/AztecProtocol/aztec-packages/issues/7071#issuecomment-2178350190 for context).

I've noticed that we have a similar if-else [here](https://github.com/AztecProtocol/aztec-packages/blob/52ce25d1abcc5a8cff7ec360acf23806cb317b57/noir-projects/aztec-nr/aztec/src/note/utils.nr#L43) and there is a comment that the if-else will go away once we inject nonces in public kernel (https://github.com/AztecProtocol/aztec-packages/issues/1386).

Is it the case here as well?

Reply:
Yep we won'y need the if statement here when publicly created notes have nonces! I think for now, instead of always hashing a nonce even when it's zero, which will require changing things elsewhere, we can compute the inner hash first. And only further hash the nonce to it if it's not zero:
```
let mut hash = compute_inner_note_hash(note);
if header.nonce != 0 {
  hash = compute_unique_hash(header.nonce, hash);
}
hash
```